### DISCUSSION TITLE
Reinit with different descriptors (to enable/disable certain USB classes at later points in time)

### DISCUSSION THREAD

**Discussion by snellejelle99** [hathach/tinyusb#2752] [Q&A]
Hi,

I am looking at a use case where I have a USB CDC and MSC device.
However instead of both enumerating at the same time I want to first enumerate as CDC only and then later enable MSC.
(preferably without breaking the CDC connection but that is not a hard requirement.)

Is there an example or has anyone done this before?
Can I just reinit tinyusb with a different set of descriptors?
If so is there anything specific I should watch out for?

All help is welcome.
 :)

**Comment by pstadelmann** [comment]
Hi,

I did something similar with CDC first, then CDC + UVC (see https://github.com/hathach/tinyusb/discussions/2717). It's enough to issue tud_disconnect / tud_connect to make it work, however this interrupts the CDC communication. This not ideal but I couldn't find another way to force the host to notice the new interface. From what I understand, this is not a limitation of TinyUSB, but it's caused by the way enumeration works in the USB protocol.
